test/parallel/test-https-server-keep-alive-timeout.js

```diff
@@ -60,29 +58,29 @@ test(function serverKeepAliveTimeoutWithPipeline(cb) {
       c.write('GET /2 HTTP/1.1\r\nHost: localhost\r\n\r\n');
       c.write('GET /3 HTTP/1.1\r\nHost: localhost\r\n\r\n');
     });
-    setTimeout(() => {
-      server.close();
-      if (socket.destroyed) destroyedSockets++;
-      cb();
-    }, 1000);
+    const interval = setInterval(() => {
+      if (socket && socket.destroyed && requestCount === 3) {
+        clearInterval(interval);
+        server.close();
+        cb();
+      }
+    }, 1);
   }));
 });
 
 test(function serverNoEndKeepAliveTimeoutWithPipeline(cb) {
   let socket;
-  let destroyedSockets = 0;
   let timeoutCount = 0;
   let requestCount = 0;
   process.on('exit', () => {
     assert.strictEqual(timeoutCount, 1);
     assert.strictEqual(requestCount, 3);
```

Comment:
I left it in to make sure it doesn't subsequently increase to 4 or higher. Probably not needed but also maybe doesn't hurt?